lib/src/utils/build_inputs.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:path/path.dart' as p;
+
+/// Builds and returns the list of inputs on which the formatter should be run.
+///
+/// These inputs are determined by expanding the [include] globs and filtering
+/// out any paths that match the expanded [exclude] globs.
+///
+/// Logs may be output in certain scenarios for debugging purposes.
+///
+/// By default these globs are assumed to be relative to the current working
+/// directory, but that can be overridden via [root] for testing purposes.
+Map<String, dynamic> buildInputs(
+    {List<Glob> exclude, List<Glob> include, String root}) {
+  Map<String, dynamic> returnInputs = {
+    'files': <String>[],
+    'skipped': {
+      'links': <String>[],
+      'excluded': <String>[],
+      'hidden_directories': Set<String>(),
+    },
+  };
+
+  exclude ??= <Glob>[];
+  include ??= <Glob>[];
+  Directory dir = Directory(root ?? '.');
+  for (final entry in dir.listSync(recursive: true, followLinks: false)) {
+    String relative = p.relative(entry.path, from: dir.path);
+    bool isExcluded = false;
+
+    if (entry is Link) {
+      returnInputs['skipped']['links'].add(relative);
+      continue;
+    }
+
+    for (final glob in exclude) {
+      if (glob.matches(relative)) {
+        returnInputs['skipped']['excluded'].add(relative);
+        isExcluded = true;
+        continue;
+      }
+    }
+    if (isExcluded) continue;
+
+    if (entry is! File || !entry.path.endsWith('.dart')) continue;
+
+    // If subdirectory starts with ".", ignore it.\
+    if (relative.startsWith('.')) {
```

Comment:
This will only catch paths that have a hidden directory as the first part of the path, e.g. `.dart_tool/build/foo.dart`, but it wouldn't skip something like `example/.pub/foo.dart`.

You can see how the dartfmt tool handles this here: https://github.com/dart-lang/dart_style/blob/f4d918c4e8e68faee2fac902c868dc368f46f497/lib/src/io.dart#L38-L46